### DISCUSSION TITLE
Geom Line dengan warna yang berbeda

### DISCUSSION THREAD

**Discussion by DeanAlexander27** [teamalgoritma/community#546] [Q&A] [answered]
Dear Kaka Instruktur
Mohon berikan referensi mengenai gabungan 2 grafik berikut menjadi 1 grafik gabungan seperti berikut contoh ...
![image](https://user-images.githubusercontent.com/123348110/219871595-f163c552-bdaa-4524-98c4-2c3b8966bda5.png)

untuk case ini sudah saya usahakan membedakan warna antara geomline dengan 2 dataframe yang berbeda tetapi selalu diambil warna yang terakhir untuk dipukul rata ke semua grafik line.. apakah ada solusi cara membedakannya berdasarkan 2 dataframe ?

ggplot() +
  geom_col(data = datasettotalstock1, aes(x = month, y = total, fill = as.factor(year)), position = 'dodge') +
  scale_fill_manual(values = c('#d0cdd7', '#acb0bd', '#416165', '#0b3948')) + 
  
  geom_line(data = datasetsales1, aes(x = month, y = total, color = factor(year), group = year), size = 1) +
  scale_colour_manual(values = c("#fe8181", "#fe5757", "#fe2e2e", "#b62020")) +
  
  geom_line(data = datasetmandist1, aes(x = month, y = total, color = factor(year), group = year), size = 1) +
  scale_color_brewer(palette = "YlOrRd") +
  
  geom_point(data = datasetsales1, aes(x = month, y = total, color = factor(year)), size = 2) +
  
  labs(x = "Month", y = "Total", title = "PT. A", subtitle = "Comparison Year on Year") +
  theme(legend.position = 'None') + theme_algoritma

![image](https://user-images.githubusercontent.com/123348110/219936021-93ebf3f1-a8c0-4078-b0e8-db7c0352b400.png)

terima kasih dan sehat selalu

**Comment by VictorNugraha** [comment]
Dear @DeanAlexander27,

Merujuk pada konsep ggplot yaitu _layer by layer_, maka warna yang akan diterapkan pada setiap garis line adalah warna yang diletakan paling bawah atau yang diterapkan pada fungsi `scale_color_brewer()`.

Mungkin bisa dicoba, mengganti fungsi `scale_color_brewer()` dengan fungsi `scale_fill_manual()` agar bisa di define satu per satu warnanya sesuai dengan keinginan Bapak. Selamat mencoba dan semoga berhasil.

Best Regards,

Team Algoritma

**Reply by DeanAlexander27** [reply]
ternyata hasilnya tetap sama ka instruktur.. apa ada case seperti ini sebelumnya ?

**Comment by delvia12** [comment]


**Reply by divakartika** [reply]
Dear @delvia12 ,

Silakan untuk membuat New Discussion terkait pertanyaan ini, sembari kami mencarikan solusi terkait hal ini, terima kasih :)

Best Regards,
Team Algoritma

**Comment by DeanAlexander27** [comment]
Halo Ka instruktur @divakartika

izin follow up pertanyaan saya .. apakah memang library ggplot2 ini ada
keterbetasan memgenai warna yang berbeda ? karena 1 dataframe memiliki 4
keterangan waktu

**Comment by jafarhf1** [comment, marked as answer]
Dear Pak @DeanAlexander27 ,

Untuk bisa mengatur color (`scale_color_*`) pada masing-masing `geom_line()`, Bapak bisa coba untuk memanfaatkan bantuan dari library `ggnewscale` kemudian menambahkan sebuah layer dengan fungsi `new_scale_color()` di antara dua `geom_line()`. Misalnya pada kasus Pak  Dean, maka bisa diletakkan seperti ini:
```
geom_line(data = datasetsales1, aes(x = month, y = total, color = factor(year), group = year), size = 1) +
scale_colour_manual(values = c("#fe8181", "#fe5757", "#fe2e2e", "#b62020")) +

new_scale_color() +

geom_line(data = datasetmandist1, aes(x = month, y = total, color = factor(year), group = year), size = 1) +
scale_color_brewer(palette = "YlOrRd") +
```

Demikian yang dapat kami sampaikan, Semoga membantu

**Reply by DeanAlexander27** [reply]
Dear Ka [@jafarhf1](https://github.com/jafarhf1)

Ah baik ka.. terima kasih atas insight nya ka.. segera saya coba..

Best regards
Dean Alexander

**Reply by DeanAlexander27** [reply]
wah bekerja ka.. terima kasih.. sangat membantu.... ka @jafarhf1

**Reply by DeanAlexander27** [reply]
dear ka @jafarhf1 
untuk hasil sudah sesuai ka 
![image](https://user-images.githubusercontent.com/123348110/220271785-91a5afaf-57e9-4935-8473-22e0637adc4b.png)

tetapi kenapa ketika di ggplotly hanya diambil line terakhir ya ?

![image](https://user-images.githubusercontent.com/123348110/220271896-af429c85-dfed-4651-b04b-16e7aaf8ca77.png)

berikut koding nya 

a <- datasettotalstock1 %>%
  ggplot() +
    geom_col(data = datasettotalstock1,mapping = aes(x = month, y = total, fill = as.factor(year), text=label),position = 'dodge') +
  scale_fill_manual(name = NULL ,values = c('#d0cdd7', '#acb0bd', '#416165', '#0b3948')) +
  geom_line(data = datasetmandist1, aes(x = month, y = total, color = factor(year), group = year), size = 1) +
  geom_point(data = datasetmandist1, mapping = aes(x = month, y = total, color = factor(year),text=label)) +
  scale_color_brewer(palette = "Set3") +
  new_scale_color() +
  
  geom_line(data = datasetsales1, aes(x = month, y = total, color = factor(year), group = year), size = 1) +
  geom_point(data = datasetsales1, mapping = aes(x = month, y = total, color = factor(year),text=label)) +
  scale_colour_manual(values = c("#fe8181", "#fe5757", "#fe2e2e", "#b62020")) +
  theme_algoritma
a

ggplotly(a,tooltip='text')

**Reply by jafarhf1** [reply]
Dear Pak @DeanAlexander27 ,

Sebelumnya mohon maaf karena respon yang telat. Sepertinya fungsi `ggplotly()` belum support dengan penggunaan `ggnewscale`  sehingga dia hanya membaca line yang terakhir saja. Sebagai saran,  Bapak perlu terlebih dahulu untuk menggabungkan kedua dataframenya. berikut adalah beberapa tahapan yang bisa Bapak lakukan:
1. Membuat kolom baru di masing-masing dataframe berisi kategori untuk menandakan bahwa nilai total adalah total untuk **datasetsales1** dan  total untuk **datasetmandist1**.
2. Menggabungkan kedua dataframe tersebut menggunakan fungsi `bind_rows()` dari package `dplyr`
3. Membuat plot line yang dibedakan berdasarkan dua kolom kategori yaitu kategori tahun dan kategori yang tadi dibuat. Bapak bisa melihat referensi berikut: [Referensi](https://stackoverflow.com/questions/44333757/plotting-color-based-on-the-combination-of-two-column-levels) 
4. Alternatif lain, Bapak bisa juga membedakannya berdasarkan 2 parameter yang berbeda misalnya untuk tahun bisa menggunakan `color` sedangkan untuk membedakan sales dan mandist bisa menggunakan `linetype`

Demikian yang dapat kami sampaikan, Semoga membantu

Best Regards,
Team Algoritma

**Reply by DeanAlexander27** [reply]
wah mantap ka @jafarhf1 ...
saya sudah ikut yang nomor 4 .. terima kasih
